.circleci/config.yml

```diff
@@ -30,7 +30,7 @@ jobs:
           echo "$WEBAPP_GIT_COMMIT"
 
           FILE_DIST=dist.tar.gz
-          until curl --max-time 75 -f -o $FILE_DIST https://pr-builds.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz; do echo waiting for webapp build; sleep 5; done;
+          until curl --retry 75 --retry-delay 5 --retry-max-time 375 -f -o $FILE_DIST https://pr-builds.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz; do echo waiting for webapp build; done;
```

Comment:
I think we should add a `--retry-all-errors` and completely drop the `until` given curl itself will loop the given number of times.

Reply:
I don't have `--retry-all-errors` on my machine yet (since curl 7.71.0). I doubt this build image has it. I guess transient errors are fine for now. Sure it can be improved later on. 
I removed the until. Curl is amazing.

Reply:
Is 403 considered a transient error? I believe that's what's returned. Just making sure it doesn't exit at first attempt ehe

Reply:
Yipyip!
```
Transient error means either: a timeout, an FTP 4xx response code or an HTTP 408 or 5xx response code.
```
From the manpage.

Reply:
OK, wait. I am wrong again.

Reply:
![Screenshot from 2021-09-06 10-39-32](https://user-images.githubusercontent.com/9913860/132190410-fa0425ee-cc00-4d8d-98f9-d11e97f0f503.png)
We need the until and --max-time until we can get a newer curl version.

Reply:
This could also explain why the trap before failed with exit 22.